Jenkinsfile

```diff
@@ -34,25 +41,29 @@ import java.util.regex.Pattern
  * - https://plugins.jenkins.io/lockable-resources
  * - https://plugins.jenkins.io/pipeline-github
  * - https://plugins.jenkins.io/email-ext
+ * - https://plugins.jenkins.io/config-file-provider
+ * - https://plugins.jenkins.io/pipeline-utility-steps
  *
  * Also you might need to allow the following calls in <jenkinsUrl>/scriptApproval/:
  * - method java.util.Map putIfAbsent java.lang.Object java.lang.Object
  * - staticMethod org.jenkinsci.plugins.pipeline.modeldefinition.Utils markStageSkippedForConditional java.lang.String
  * - new java.lang.IllegalArgumentException java.lang.String
  * - method hudson.plugins.git.GitSCM getUserRemoteConfigs
  * - method hudson.plugins.git.UserRemoteConfig getUrl
+ * - method java.lang.Throwable addSuppressed java.lang.Throwable
  *
  * Just run the script a few times, it will fail and display a link to allow these calls.
  */
 
 @Field final String MAVEN_LOCAL_REPOSITORY_RELATIVE = '.repository'
 @Field final String MAVEN_TOOL = 'Apache Maven 3.5.2'
 
-@Field final Pattern MAIN_REPOSITORY_URL_PATTERN = ~/^(git@github.com:|https:\/\/github.com\/)hibernate\/.*.git$/
-// The following should be a space-separated list of email addresses
-@Field final String MAIN_REPOSITORY_MAINTAINER_EMAILS = "yoann+hibernate-ci@hibernate.org guillaume+ci@hibernate.org"
-
+// Default node pattern, to be used for resource-intensive stages.
+// Should not include the master node.
 @Field final String NODE_PATTERN_BASE = 'Slave'
+// Quick-use node pattern, to be used for very light, quick, and environment-independent stages,
+// such as sending a notification. May include the master node in particular.
+@Field final String QUICK_USE_NODE_PATTERN = 'Master||Slave'
```

Comment:
I don't understand why we would need that? Shouldn't the configuration file parsing be done on the master before distributing the job to the slaves?

Reply:
Previously the configuration stage did not have a `node` statement, meaning everything indeed happened on the master node, but some features were not available. In particular, filesystem access is disabled. You have to allocate a node explicitly for that. Which is why I added a `node()` statement just to fetch the configuration file.

I'd rather not require to use a slave node for that, since they might all be busy, and it would be a shame to wait for a node just to access a configuration file.

As long as we don't use the master node for anything else than very short operations, that should not slow down the build (it just takes one or two seconds).

You can try without the `node` statement if you want, you'll just get an exception telling you that some context is missing.

Reply:
Well, what's bothering is more than we could execute this on a slave. I think I would have made it master only.

Anyway, we can merge it.

Reply:
It should work on a slave as well. The configuration files are accessible on any node, that's the point of this plugin.
Merging, thanks.